app/components/autocomplete/at_mention/at_mention.js

```diff
@@ -50,74 +50,74 @@ export default class AtMention extends Component {
     constructor(props) {
         super(props);
 
-        const ds = new ListView.DataSource({
-            sectionHeaderHasChanged: (s1, s2) => s1 !== s2,
-            rowHasChanged: (r1, r2) => r1 !== r2
-        });
-        const data = {};
         this.state = {
-            active: false,
-            dataSource: ds.cloneWithRowsAndSections(data)
+            sections: []
         };
     }
 
     componentWillReceiveProps(nextProps) {
-        const {isSearch} = nextProps;
-        const regex = isSearch ? FROM_REGEX : AT_MENTION_REGEX;
-        const match = nextProps.postDraft.substring(0, nextProps.cursorPosition).match(regex);
+        const {autocompleteUsers, hasMatch, isSearch, matchTerm, requestStatus} = nextProps;
 
-        if (!match || this.state.mentionComplete) {
+        if (!hasMatch || this.state.mentionComplete) {
```

Comment:
Can we change this to check if `nextProps.hasMatch !== this.props.hasMatch`? Right now, it's clearing the state more than it needs to, and causing a re-render because `this.state.sections` is changing each time